app/services/photo_service.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+class PhotoService
+  def initialize(user=nil, deny_raw_files=true)
+    @user = user
+    @deny_raw_files = deny_raw_files
+  end
+
+  def visible_photo(photo_guid)
+    Photo.owned_or_visible_by_user(@user).where(guid: photo_guid).first
+  end
+
+  def create_from_params_and_file(base_params, uploaded_file)
```

Comment:
- Metrics/AbcSize: Assignment Branch Condition size for create_from_params_and_file is too high. [20.54/20]
- Metrics/CyclomaticComplexity: Cyclomatic complexity for create_from_params_and_file is too high. [7/6]